src/metrics.jl

```diff
@@ -144,7 +144,24 @@ SqEuclidean() = SqEuclidean(0)
 #
 ###########################################################
 
-function evaluate(d::UnionMetrics, a::AbstractArray, b::AbstractArray)
+# Specialized for Arrays and avoids a branch on the size
+@inline function evaluate(d::UnionMetrics, a::T, b::T) where T <: Union{Array, UnsafeVectorView}
```

Comment:
Note this `@inline`. It apparently does quite a lot for small arrays and it isn't that much code to inline.